packages/rule-typescript-config/src/target.ts

```diff
@@ -0,0 +1,235 @@
+         */
+        const isOlder = (version1: number | string, version2: number | string): boolean => {
+
+            if (typeof version1 === 'number' && typeof version2 === 'number') {
+                return version1 < version2;
+            }
+
+            // Can't store `typeof` or otherwise TypeScript complains about `includes` not being on `number`
+            if (typeof version1 === 'string' && typeof version2 === 'string') {
+                if (version1.includes('-') || version2.includes('-')) {
+                    const range1 = version1.split('-')[0];
+                    const range2 = version2.split('-')[0];
+
+                    if (range1.includes('.') && range2.includes('.')) {
+                        const parts1 = range1.split('.');
+                        const parts2 = range2.split('.');
+
+                        let older = true;
+
+                        for (let i = 0; i < parts1.length && older; i++) {
+                            older = isOlder(parseInt(parts1[i]), parseInt(parts2[i]));
+                        }
+
+                        return older;
+
+                    }
+
+                    return isOlder(parseInt(range1), parseInt(range2));
+                }
+            }
+
+            // If we reach this point we are probably comparing something like "android 62" and "android 4.4.3-4.4.4"
+            return isOlder(getMajor(version1), getMajor(version2));
+        };
+
+        /** Transforms a browserslist array into an object with the minimum versions of each one */
+        const toMiniumBrowser = (targetedBrowsers: Array<string>) => {
+            const configuration = targetedBrowsers.reduce((config, browserVersion) => {
+                // The format in the array is "browser version". E.g.: "firefox 52"
+                const [browser, strVersion] = browserVersion.split(' ');
+                const version = strVersion.includes('.') || strVersion.includes('-') ? strVersion : parseInt(strVersion);
+                const previousVersion = config[browser];
+
+
+                if (!previousVersion || isOlder(version, previousVersion)) {
+                    config[browser] = version;
+                }
+
+                return config;
+            }, {});
+
+            return configuration;
+        };
+
+        /**
+         * Based on the minimum supported browser configuration passed,
+         * determines what's the maximum ES version that can be targetted.
+         */
+        const getMaxVersion = (minimumBrowsers): string => {
+            const versions = Object.keys(compatMatrix);
+            let maxVersion = 'es3';
+
+            /**
+             * This will check all the ES versions and compare all the
+             * browsers from the compat matrix agains the given ones.
+             * If there's one in the compat matrix newer than the ones
+             * provided, it will fail.
+             */
+            versions.forEach((version) => {
```

Comment:
shouldn't  this be a `reduce`?

Reply:
Yes, I can probably change it to `reduce`